pages/item/index.js

```diff
@@ -10,125 +10,137 @@ import BreadcrumbsModule from "components/ItemComponents/BreadcrumbsModule";
-            className={css.citeButton}
-            toCiteText="item"
-            title={item.title}
-          />
-          <CheckableLists itemId={item.id} />
-        </div>
-      </div>
+                <div className={css.faveAndCiteButtons}>
+                    <CiteButton
+                        creator={item.creator}
+                        displayDate={item.date ? item.date.displayDate : item.date}
+                        spatialName={item.spatial ? item.spatial.name : item.spatial}
+                        sourceUrl={item.sourceUrl}
+                        className={css.citeButton}
+                        toCiteText="item"
+                        title={item.title}
+                    />
+                    <CheckableLists itemId={item.id}/>
+                </div>
+            </div>
 
-    </MainLayout>
-  );
+        </MainLayout>
+    );
 };
 
-ItemDetail.getInitialProps = async context => {
-  const query = context.query;
-  const req = context.req;
-  const res = context.res;
-  const isQA = parseCookies(context).hasOwnProperty("qa");
-  const currentFullUrl = getCurrentFullUrl(req);
-  const currentUrl = getCurrentUrl(req);
-  const randomItemId = isQA ? await getRandomItemIdAsync(currentUrl) : null;
-  // check if item is found
-  try {
-    const res = await fetch(`${currentUrl}${API_ENDPOINT}/${query.itemId}`);
-    const json = await res.json();
-    const doc = json.docs[0];
-    const thumbnailUrl = getItemThumbnail(doc);
-    const date = doc.sourceResource.date &&
-      Array.isArray(doc.sourceResource.date)
-      ? doc.sourceResource.date[0]
-      : doc.sourceResource.date;
-    const language = doc.sourceResource.language &&
-      Array.isArray(doc.sourceResource.language)
-      ? doc.sourceResource.language.map(lang => {
-          return lang.name;
-        })
-      : doc.sourceResource.language;
-    const strippedDoc = Object.assign({}, doc, { originalRecord: "" });
-    delete strippedDoc.originalRecord;
-    return {
-      currentFullUrl,
-      item: Object.assign({}, doc.sourceResource, {
-        id: doc.id,
-        thumbnailUrl,
-        contributor: doc.dataProvider,
-        intermediateProvider: doc.intermediateProvider,
-        date: date,
-        language: language,
-        partner: doc.provider.name,
-        sourceUrl: doc.isShownAt,
-        useDefaultImage: !doc.object,
-        edmRights: doc.rights,
-        doc: strippedDoc,
-        originalRecord: doc.originalRecord
-      }),
-      randomItemId,
-      isQA
-    };
-  } catch (error) {
-    console.log(error);
-    if (res) {
-      res.statusCode = 404;
+export const getServerSideProps = async context => {
+    const query = context.query;
+    const req = context.req;
+    const res = context.res;
+    const isQA = parseCookies(context).hasOwnProperty("qa");
+    const currentFullUrl = `${req.protocol}://${req.get("host")}${req.url}`;
+    const currentUrl = `${req.protocol}://${req.get("host")}`;
```

Comment:
These string concatenations for `currentUrl` and `currentFullUrl` are showing up a lot.  Would it be worth making a helper function so they don't have to be typed out every time?